Assets/Scripts/PilotController.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PilotController : MonoBehaviour
+{
+
+    private bool isJumping;
+    private Vector3 groundCheck;
+    private bool isGrounded;
+    private LayerMask layerMask;
+
+    public float jumpForce;
+    public float frontAcceleration;
+    public float neutralAcceleration;
+    public float brakeAcceleration;
+    public float sidesAcceleration;
+    public float maxSpeed;
+    public float minSpeed;
+    public float airControlMultiplier;
+    public float Velocity
+    {
+        get
+        {
+            return Mathf.Sqrt(Mathf.Pow(rb.velocity.x, 2) + Mathf.Pow(rb.velocity.z, 2));
+        }
+    }
+
+    private bool cruiseControl;
+    private float cruiseControlSpeed;
+
+
+    private Rigidbody rb;
+
+    private void Awake()
+    {
+        rb = GetComponent<Rigidbody>();
+    }
+
+    private void Start()
+    {
+        isJumping = false;
+        isGrounded = false;
+        groundCheck = Vector3.down * 0.5f;
+        layerMask = (1 << LayerMask.NameToLayer("Ground"));
+
+        frontAcceleration = 25f;
+        neutralAcceleration = 2.5f;
+        brakeAcceleration = 25f;
+        sidesAcceleration = 50f;
+        maxSpeed = 30f;
+        minSpeed = 5f;
+        airControlMultiplier = 0.25f;
+
+        cruiseControl = false;
+        cruiseControlSpeed = minSpeed;
+    }
+
+    private void Update()
+    {
+        isGrounded = Physics.Linecast(transform.position, transform.position + groundCheck, layerMask);
```

Comment:
Can't [this](https://docs.unity3d.com/ScriptReference/CharacterController-isGrounded.html) apply instead?